controllers/horizontal_runner_autoscaler_webhook.go

```diff
@@ -586,6 +658,9 @@ HRA:
 
 			// Ensure that the RunnerSet-managed runners have all the labels requested by the workflow_job.
 			for _, l := range labels {
+				if l == "self-hosted" {
```

Comment:
I recently learned that this isn't quite true. `self-hosted` is included in a `workflow_job` event because you had included it in `runs-on` of your workflow definition yaml. So if we were to make it clear, you'd better do 

```suggestion
				if len(rs.Spec.Labels) == 0 && l == "self-hosted" {
```

So that it's clearer that actions-runner-controller just assumes the default RunnerSet.Spec.Labels and RunnerDeployment.Spec.Labels as `["self-hosted"]` if empty.

Reply:
yes - that's right `self-hosted` is included in `workflow_job`. However the proposed approach doesn't cover other examples, e.g if I have a workflow with `runs-on: ["self-hosted", "group-a"]` and I have `RunnerDeployment.Spec.Labels = ["group-a"]`, then job won't be scheduled. The only way will be to define `RunnerDeployment.Spec.Labels = ["self-hosted", "group-a"]` although that seems weird for two reasons:
1. all runners in this controller are self-hosted - if you register a runner without self-hosted in Labels it will be useless as it will never match a self-hosted runner.
2. `self-hosted` is a default label added by actions/runner which is not specified during registration - I believe in `Spec.Labels` we should set the custom labels that are going to be passed via config.sh which is not the case for self-hosted.

Any thoughts?

Reply:
@fgalind1 Thanks for helping me shape this!

I believe this discussion relates to #953 which stems from the same thought as yours.

> self-hosted is a default label added by actions/runner which is not specified during registration

So the biggest thing I missed was this one!

I was blindly assuming that `self-hosted` is just a standard label that everyone uses just by convention. If this is the hard-coded default used by the actions/runner agents as well, actions-runner-controller should treat it as such too.

I'm going to merge #953 first as it's submitted earlier and after that, I'm fine with this change being included in this PR!